torch_xla/csrc/ops/native_dropout.cpp

```diff
@@ -0,0 +1,39 @@
+#include "torch_xla/csrc/ops/native_dropout.h"
+
+#include "torch_xla/csrc/helpers.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/xla_lower_util.h"
+
+namespace torch_xla {
+namespace {
+
+xla::Shape NodeOutputShape(const torch::lazy::Value& input) {
+  xla::Shape input_shape = GetXlaShape(input);
+  input_shape.set_element_type(xla::PrimitiveType::U8);
+  xla::Shape mask_shape = GetXlaShape(input);
+  mask_shape.set_element_type(xla::PrimitiveType::PRED);
+  return xla::ShapeUtil::MakeTupleShape({input_shape, mask_shape});
+}
+
+}  // namespace
+
+NativeDropout::NativeDropout(const torch::lazy::Value& input,
+                             const torch::lazy::Value& seed, float p,
+                             c10::optional<bool> train)
+    : XlaNode(torch::lazy::OpKind(at::aten::native_dropout), {input, seed},
+              [&]() { return NodeOutputShape(input); }, 2),
```

Comment:
The hash is missing in the XlaNode constructor. You need it to distinguish different nativeDropout nodes. 

Just do
```
XlaNode(..., torch::lazy::MHash(p, train))
```

Reply:
I guess that's what's happening, if we don't have a hash for the nativedropout node, first a `train=false` node is created, then when the other test creates a node `train=true`, it will consider the `train=false` node to be the same node, and reused it.

Reply:
Thanks, I thought the default hash function will work... Could you give more details about why the default hash causes the weird status issue?

Reply:
The constructor has a default value for the hash value. see https://github.com/pytorch/xla/blob/00272c34aad2aaf61b37909ce11453c8be58aad9/torch_xla/csrc/ir.h

Reply:
Thanks for the explanation!